lib/index.js

```diff
@@ -1097,6 +1101,11 @@ vmadm.stop = function vmStop(opts, callback) {
         args.push('-F');
     }
 
+    if (opts.timeout) {
+        // all args are required to be strings
+        args.push('-t', opts.timeout.toString());
+    }
+
```

Comment:
On 3/4/16 11:12 , Josh Wilsdon wrote:
>> @@ -1097,6 +1101,11 @@ vmadm.stop = function vmStop(opts, callback) {
>>          args.push('-F');
>>      }
>>  
>> +    if (opts.timeout) {
>> +        // all args are required to be strings
>> +        args.push('-t', opts.timeout.toString());
>> +    }
>> +
> 
> vmadm ignores the option when it's not supported. In order to tell if it can be honored here, we'd need to do yet another VM.load() which would slow down every stop action. If you think it's worth it, we can do so.

Without understanding the broader context, it's hard to say whether its
worth it. From an abstract point of view, the fact that I can ask for
something and have no way of knowing if it was valid or not, seems
unfortunate.

What are the guarantees that we're trying to make? Given the way that
vmadm seems designed here, it definitely limits our options as it cannot
provide that feedback.

Robert